pages/hold/request/[id]/index.tsx

```diff
@@ -85,40 +94,44 @@ export default function HoldRequestPage({
       const response = await fetch(`${BASE_URL}/api/hold/request/${holdId}`, {
         method: "POST",
         body: JSON.stringify({
-          patronId: patronId.value,
-          source: source.value,
-          pickupLocation: pickupLocation.value,
+          patronId: patronId?.value,
+          source: source?.value,
+          pickupLocation: pickupLocation?.value,
           jsEnabled: true,
         }),
       })
       const responseJson = await response.json()
 
-      if (response.status !== 200) {
-        console.error(
-          "HoldRequestPage: Error in hold request api response",
-          responseJson.error
-        )
-        setAlert(true)
-        setErrorDetail(responseJson?.error?.detail || "")
-        setFormPosting(false)
-        return
+      switch (response.status) {
+        // Patron is ineligible to place holds
+        case 401:
+          setFormPosting(false)
+          setErrorStatus("patronIneligible")
+          setPatronEligibilityStatus(responseJson?.patronEligibilityStatus)
+          break
+        // Server side error placing the hold request
+        case 500:
+          setFormPosting(false)
```

Comment:
Here's a suggestion a la Refactoring...
You can pull out the invocations within each case to functions that summarize the comments above them. That way, you can also reuse a `serverSideErrorPlacingHoldRequest` call  in the catch block on line 115.